plugins/single-instance/src/platform_impl/macos.rs

```diff
@@ -4,13 +4,164 @@
+                    eprintln!("single_instance failed - launching normally: {}", e);
+                }
+                None => {
+                    // Not an OS error -> Try to launch instead
+                    eprintln!("single_instance failed - launching normally: {}", err);
+                }
+            }
+        }
+    };
 }
 
-pub fn destroy<R: Runtime, M: Manager<R>>(_manager: &M) {}
+fn listen_for_other_instances<A: Runtime>(
+    listener: UnixListener,
+    app: AppHandle<A>,
+    mut cb: Box<SingleInstanceCallback<A>>,
+) {
+    let cwd = std::env::current_dir()
+        .unwrap_or_default()
+        .to_str()
+        .unwrap_or_default()
+        .to_string();
+
+    tauri::async_runtime::spawn(async move {
+        for stream in listener.incoming() {
+            match stream {
+                Ok(mut stream) => {
+                    let mut s = String::new();
+                    match stream.read_to_string(&mut s) {
+                        Ok(_) => {
+                            let args: Vec<String> = s.split('\0').map(String::from).collect();
+                            cb(&app.clone().app_handle(), args, cwd.clone());
+                        }
+                        Err(e) => println!("Error reading stream: {e}"),
+                    }
+                }
+                Err(err) => {
+                    eprintln!("single_instance notify failed: {}", err);
+                    continue;
+                }
+            }
+        }
+    });
+}
+
+fn notify_singleton(socket: PathBuf) -> Result<(), ErrorKind> {
+    match UnixStream::connect(&socket) {
+        Ok(stream) => {
+            let mut bf = BufWriter::new(&stream);
+            let args_joined = std::env::args().collect::<Vec<String>>().join("\0");
+
+            if let Err(_e) = bf.write_all(args_joined.as_bytes()) {
+                return Err(ErrorKind::Abort);
+            }
+            if let Err(_e) = bf.flush() {
+                return Err(ErrorKind::Abort);
+            }
+
+            drop(bf);
+            Ok(())
+        }
+        Err(e) => {
+            match e.raw_os_error() {
+                Some(61) => {
+                    // An old singleton likely did not clean up socket properly.
+                    // -> Delete old socket and retry.
+                    socket_cleanup(socket);
+                    Err(ErrorKind::Retry)
+                }
+                Some(_) => {
+                    // Unexpected OS error -> Try to launch instead.
+                    Err(ErrorKind::Abort)
+                }
+                None => {
+                    // Not an OS error -> Try to launch instead
+                    Err(ErrorKind::Abort)
+                }
+            }
+        }
+    }
+}
+
+fn socket_path(config: &Config) -> PathBuf {
+    let identifier = config.identifier.replace(['.', '-'].as_ref(), "_");
+    // Use /tmp as socket path must be shorter than 100 chars.
+    PathBuf::from(format!("/tmp/{}_si.sock", identifier))
```

Comment:
Some input here would be appreciated on the usage of the `/tmp` directory. The character limit on socket files prevents usage of Tauri's suggested directories. I also tried `/var/run` as that's where docker.sock is located but that also gives errors.

Reply:
/tmp is fine afaik, that's what i used in other tauri projects too and it didn't cause any issues.